src/Routing/Route/EntityRoute.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Routing\Route;
+
+use ArrayAccess;
+use RuntimeException;
+
+/**
+ * Matches entities to routes
+ *
+ * This route will match by entity and map its fields to the URL pattern by
+ * comparing the field names with the template vars. This makes it easy and
+ * convenient to change routes globally.
+ */
+class EntityRoute extends Route
+{
+    /**
+     * Match by entity and map its fields to the URL pattern by comparing the
+     * field names with the template vars
+     *
+     * @param array $url Array of parameters to convert to a string.
+     * @param array $context An array of the current request context.
+     *   Contains information such as the current host, scheme, port, and base
+     *   directory.
+     * @return bool|string Either false or a string URL.
+     */
+    public function match(array $url, array $context = [])
+    {
+        if (isset($url['_entity'])) {
+            $this->_checkEntity($url['_entity']);
+
+            $entity = $url['_entity'];
+            preg_match_all('@:(\w+)@', $this->template, $matches);
+
+            foreach ($matches[1] as $field) {
+                $url[$field] = $entity[$field];
```

Comment:
What if we want to pass a association key like `category_id`in route ?
Maybe something like this : 
```php
$url[$field] = $url['_entity'][\Cake\Utility\Inflector::delimit($field)];
```

Reply:
`Inflector::delimit()` converts a CamelCased string to underscored, which field name already are by convention. So why use `delimit()`?

Reply:
Route could be like this :
```php
$routes->connect('/posts/:categoryId/:slug-:id', ['controller' => 'Posts', 'action' => 'category']);
```
If not use `delimit()`, we need to catch the underscore in the regex

Reply:
Oh. My bad, `\w` catch `_` too.